### PR TITLE
Fix psglib warning messages

### PR DESCRIPTION
**Pull request by DanIverson**
With newer gcc compilers, a lot of warnings are generated from pulse
sequence compilations (psglib). Fixed these but tried to avoid changing
any software logic.

### CODE REVIEW

File: src/biopack/psg/CompSens.h [hunk 1]
```diff
@@ -46,7 +47,7 @@ int    xni, xni2, xni3;
     psg_abort(1);
   }
 
-  fgets(cmd, 512, src);
+  ret = fgets(cmd, 512, src);
```
Comment (tesch1) [inline src/biopack/psg/CompSens.h]: the 512 could be sizeof(cmd)

File: src/biopack/psg/CompSens.h [hunk 2]
```diff
@@ -138,6 +140,7 @@ char *opt;        // reserved for weighted sampling modes
     sprintf(fname,"%s/sampling.sch", curexp);
     if((stype[0] =='a') || (stype[0] == 'y'))
     {
+      int ret __attribute__((unused));
```
Comment (tesch1) [inline src/biopack/psg/CompSens.h]: These silence the compiler warning, but dont really address the issue that the compiler warning is meant to alert us to.  An equivalent option would be disabling the compiler warning until we can test things thoroughly enough to know whether code changes (detecting and handling failures) is going to break anything upstream from this.

File: src/biopack/psg/CompSens.h (hunk 1 above)
Comment (DanIverson) [reply src/biopack/psg/CompSens.h]: Agreed. For this PR, I tried to make only the changes needed to silence the warnings. While reviewing the code I found a number of other infelicities in the code but I resisted making changes. A fairly serious one is a potential endless loop in sglCommon.c (lines 6715-6718)

File: src/biopack/psg/CompSens.h (hunk 2 above)
Comment (DanIverson) [reply src/biopack/psg/CompSens.h]: Also agreed. However, I did not want to use -Wno-... options to silence the warnings. These files are compiled when a user compiles their own pulse sequence (at least solids, imaging, and bio sequences) and I did not want them to miss problems with their code. But I also did not want them to be distracted by problems in these "include" files. I figured we could always grep for __attribute__((unused)) to identify where real fixes should be made.

Comment (tesch1) [reply src/biopack/psg/CompSens.h]: Ah, yeah, that makes total sense.

Comment (tesch1) [reply src/biopack/psg/CompSens.h]: btw, I wrote a little log parser to tally warning counts from a log file: https://github.com/tesch1/OpenVnmrJ/blob/fix-test/scripts/grok_warnings.py

**Pull request merged**